internal/upstreamldap/upstreamldap.go

```diff
@@ -869,43 +874,22 @@ func getDomainFromDistinguishedName(distinguishedName string) (string, error) {
 	return strings.Join(domainComponents[1:], "."), nil
 }
 
-func PwdUnchangedSinceLogin(entry *ldap.Entry, attributes provider.StoredRefreshAttributes) error {
-	authTime := attributes.AuthTime
-	pwdLastSetWin32Format := entry.GetAttributeValues(PwdLastSetAttribute)
-	if len(pwdLastSetWin32Format) != 1 {
-		return fmt.Errorf("expected to find 1 value for %s attribute, but found %d", PwdLastSetAttribute, len(pwdLastSetWin32Format))
-	}
-	// convert to a time.Time
-	pwdLastSetParsed, err := win32timestampToTime(pwdLastSetWin32Format[0])
-	if err != nil {
-		return err
-	}
-
-	// if pwdLastSet > authTime, that means that the password has been changed since the initial login.
-	// return an error so the user is prompted to log in again.
-	if pwdLastSetParsed.After(authTime) {
-		return fmt.Errorf("password has changed since login. login time: %s, password set time: %s", authTime, pwdLastSetParsed)
-	}
-	return nil
-}
-
-func win32timestampToTime(win32timestamp string) (time.Time, error) {
-	// take a win32 timestamp (represented as the number of 100 ns intervals since
-	// January 1, 1601) and make a time.Time
+func AttributeUnchangedSinceLogin(attribute string) func(*ldap.Entry, provider.StoredRefreshAttributes) error {
+	return func(entry *ldap.Entry, storedAttributes provider.StoredRefreshAttributes) error {
+		prevAttributeValues := storedAttributes.AdditionalAttributes[attribute]
+		newValues := entry.GetAttributeValues(attribute)
```

Comment:
We should (after checking for single element) use `base64.RawURLEncoding.EncodeToString(entry.GetRawAttributeValues(attribute)[0])`.  LDAP attributes can be binary, and since this is meant to be a generic function, it should not assume anything about the schema of the attribute.